modules/api/src/main/java/org/apache/ignite/table/ReceiverDescriptor.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.table;
+
+import java.util.List;
+import java.util.Objects;
+import org.apache.ignite.compute.DeploymentUnit;
+
+/**
+ * Data streamer receiver descriptor.
+ */
+public class ReceiverDescriptor {
+    private final String receiverClassName;
+
+    private final List<DeploymentUnit> units;
+
+    private ReceiverDescriptor(String receiverClassName, List<DeploymentUnit> units) {
+        Objects.requireNonNull(receiverClassName);
+        Objects.requireNonNull(units);
+
+        this.receiverClassName = receiverClassName;
+        this.units = units;
+    }
+
+    /**
+     * Streamer receiver class name.
+     *
+     * @return Streamer receiver class name.
+     */
+    public String receiverClassName() {
+        return receiverClassName;
+    }
+
+    /**
+     * Deployment units.
+     *
+     * @return Deployment units.
+     */
+    public List<DeploymentUnit> units() {
+        return units;
+    }
+
+    /**
+     * Create a new builder.
+     *
+     * @return Receiver descriptor builder.
+     */
+    public static Builder builder(String receiverClassName) {
+        Objects.requireNonNull(receiverClassName);
+
+        return new Builder(receiverClassName);
+    }
+
+    /**
+     * Create a new builder.
+     *
+     * @return Receiver descriptor builder.
+     */
+    public static Builder builder(Class<? extends DataStreamerReceiver<?, ?>> receiverClass) {
+        Objects.requireNonNull(receiverClass);
+
+        return new Builder(receiverClass.getName());
+    }
+
+    /**
+     * Builder.
+     */
+    public static class Builder {
+        private final String receiverClassName;
+        private List<DeploymentUnit> units;
+
+        private Builder(String receiverClassName) {
+            Objects.requireNonNull(receiverClassName);
+
+            this.receiverClassName = receiverClassName;
+        }
+
+        /**
+         * Sets the deployment units.
+         *
+         * @param units Deployment units.
+         * @return This builder.
+         */
+        public Builder units(List<DeploymentUnit> units) {
+            this.units = units;
```

Comment:
Shall we copy List?

Reply:
We don't copy in `JobDescriptor`. Also, `units()` property exposes a mutable `List`, so we would need to copy there as well for full protection. I don't think it's worth it, let's keep as is for now.